sycl/include/CL/sycl/detail/pi.h

```diff
@@ -1769,6 +1772,14 @@ __SYCL_EXPORT pi_result piextPluginGetOpaqueData(void *opaque_data_param,
 /// \param PluginParameter placeholder for future use, currenly not used.
 __SYCL_EXPORT pi_result piTearDown(void *PluginParameter);
 
+/// API to get Plugin specific warning and error messages.
+/// \param message is a pointer to an array of characters which will be filled
+/// with the error message. \param message_size is the size of the array message
+/// points to. \param is_warning is a bool indicating if the message is a
+/// non-failing error.
+__SYCL_EXPORT pi_result piPluginGetLastError(char *message, size_t message_size,
+                                             bool *is_warning);
```

Comment:
Maybe we can not add "is_warning" but return PI_SUCCESS to indicate that upper RT should treat it as a warning , but return other PI error codes for something that is an error that SYCL RT should try to recover from or throw an exception back to the user.

Reply:
Also we need a way for SYCL RT to know the size of  `message` array. Or we can just return a pointer directly(without the copy) which can be used by the SYCL RT until the next PI call is done(assuming we take `thread_local` approach from another comment).

Reply:
I have updated it to not use `is_warning` but return a user specified code, so PI_SUCCESS is treated as a warning while others will trigger an error. 
Additionally it now uses `thread_local` so it returns a ptr to the message as suggested.